packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -0,0 +1,134 @@
+                l.replace(/\/$/, '')?.replace(overrideReplace.replace(/\//, ''), ''),
+                [end, rest]
+            )
+        })
+    }
+
+    /**
+     *
+     * @param requestPath
+     * @param dirs
+     */
+    findFileMap(requestPath, dirs) {
+        var fileExt = path.extname(requestPath)
+        for (var dir of dirs) {
+            var base = path.join(dir, requestPath)
+            if (fileExt) {
+                const noExtPath = requestPath.replace(fileExt, '')
+                if (this.extendsHashMap.has(noExtPath)) {
+                    return base
+                }
+            } else {
+                if (this.extendsHashMap.has(requestPath)) {
+                    const end = this.extendsHashMap.get(requestPath)[1]
+
+                    if (end[0] === 'index') {
+                        base = path.join(base, this.extendsHashMap.get(requestPath)[1].join(''))
+                        return base
+                    } else {
+                        base = base?.replace(/$\//, '') + end.join('')
+                        return base
+                    }
+                }
+            }
+        }
+    }
+
+    toOverrideRelative(path) {
+        var override = this.findOverride(path)
+        return path.substring(override.length + 1)
+    }
+
+    findOverride(path) {
+        return this._allSearchDirs.find((override) => {
+            return path.indexOf(override) === 0
+        })
+    }
+
+    isFromExtends(request, path) {
+        return (
+            this.extends.includes(request?.split('/')?.[0]) &&
+            // this is very important, to avoid circular imports, check that the
+            // `issuer` (requesting context) isn't the overrides directory
+            !path.match(this.projectDir + this.overridesDir)
+        )
+    }
+
+    apply(resolver) {
+        resolver.getHook('resolve').tapAsync(
+            'FeatureResolverPlugin',
+            function (requestContext, resolveContext, callback) {
+                let targetFile
+                let overrideRelative
+                if (this.isFromExtends(requestContext.request, requestContext.path)) {
+                    overrideRelative = this.toOverrideRelative(requestContext.request)?.replace(
+                        /$\//,
+                        ''
+                    )
+                    targetFile = this.findFileMap(overrideRelative, this._allSearchDirs)
+                }
+
+                if (targetFile) {
+                    const target = resolver.ensureHook('resolved')
+                    requestContext.path = targetFile
+                    resolver.doResolve(
+                        target,
+                        requestContext,
+                        `${this.constructor.name} found base override file`,
+                        resolveContext,
+                        callback
+                    )
+                } else {
+                    return callback()
+                }
```

Comment:
(super nit, remove the `else`)
```suggestion
                    )
                    return
                }
                return callback()
```

Reply:
the `if` portion doesn't return, I'm not confident this wouldn't call `callback()` twice